test/Microsoft.IdentityModel.Xml.Tests/EnvelopedSignatureWriterTests.cs

```diff
@@ -150,13 +237,11 @@ public void RoundTripWsMetadata(EnvelopedSignatureTheoryData theoryData)
                 {
                     Encoding = new UTF8Encoding(false)
                 };
+
                 var buffer = new MemoryStream();
                 var esw = new EnvelopedSignatureWriter(XmlWriter.Create(buffer, settings), theoryData.SigningCredentials, theoryData.ReferenceId);
 
-                foreach (var action in theoryData.Action.GetInvocationList())
-                {
-                    action.DynamicInvoke(esw);
-                }
+                theoryData.Action.DynamicInvoke(esw);
```

Comment:
nit: DynamicInvoke is going to be much slower than using Invoke or directly calling the delegate.

Reply:
@keegan-caruso thanks, i am on the fence about actions in general. it makes it easy to set a breakpoint on the specific test, but adds a layer.